### PR TITLE
[WIP] [Excalibur] UI polish: add tooltip, allow tooltip placement option to be passed in

### PR DESCRIPTION
**Pull request by amyrlam**
## Description of the change

Successfully adds Excalibur tooltip to the right. But do we want the tooltip to be on the top / is this feature important?

//

WIP: Add placement option for Tooltip 

With this commit, placement="right" looks same as before.

❓But placement="top" has design errors, am not sure where to fix them yet. Do in CSS or use some sort of "tethering" library that determines the position?

Will need to check placement="bottom" and placement="left" as well, if needed.

❓Not sure if possible to make placement optional in TypeScript, with "right" the default value? But limited to the available values from the Popper library.

tooltip

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes part of https://github.com/Recidiviz/covid19-dashboard/issues/300

### Development

These boxes should be checked by the submitter prior to merging:

- [ ] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/design-system/InputDate.tsx [hunk 1]
```diff
@@ -84,7 +84,11 @@ const InputDate: React.FC<InputBaseProps<Date>> = (props) => {
 
   return (
     <InputContainer>
-      <InputLabelAndHelp label={labelAbove} labelHelp={labelHelp} />
+      <InputLabelAndHelp
+        label={labelAbove}
+        labelHelp={labelHelp}
+        placement="right"
```
Comment (szhu) [inline src/design-system/InputDate.tsx:90]: Maybe `labelHelpPlacement` to clarify what it's for?

File: src/design-system/HelpButtonWithTooltip.tsx [hunk 2]
```diff
@@ -19,11 +19,27 @@ const HelpButton = styled.span`
...
+    | "top-end"
+    | "bottom"
+    | "bottom-start"
+    | "bottom-end"
+    | "right"
+    | "right-start"
+    | "right-end"
+    | "left"
+    | "left-start"
+    | "left-end";
```
Comment (szhu) [inline src/design-system/HelpButtonWithTooltip.tsx:37]: (Occurrence 1 of 3 of this)

---

Suggestion: You can directly reference the allowed types instead:

```suggestion
  placement: PopperProps["placement"]
```

This way, if a future version of Popper adds more placements in the future, we won't have an inconsistent set of allow props. I think it's also ok not reference that type and have our own list of placements, but in that case we should reduce the list to just the placements that we'll need.

---

Note: The way I found this type is by inspecting the definition of "placement" here! (Cmd-click or alt-click depending on your VSCode setting, right-click » Go to Definition)

File: src/design-system/InputLabelAndHelp.tsx [hunk 3]
```diff
@@ -8,6 +8,22 @@ interface Props {
...
+    | "top-end"
+    | "bottom"
+    | "bottom-start"
+    | "bottom-end"
+    | "right"
+    | "right-start"
+    | "right-end"
+    | "left"
+    | "left-start"
+    | "left-end";
```
Comment (szhu) [inline src/design-system/InputLabelAndHelp.tsx:26]: (Occurrence 2 of 3 of this)

File: src/design-system/Tooltip.tsx [hunk 4]
```diff
@@ -87,6 +87,22 @@ const TooltipArrowDiv = styled.div`
...
+    | "top-end"
+    | "bottom"
+    | "bottom-start"
+    | "bottom-end"
+    | "right"
+    | "right-start"
+    | "right-end"
+    | "left"
+    | "left-start"
+    | "left-end";
```
Comment (szhu) [inline src/design-system/Tooltip.tsx:105]: (Occurrence 3 of 3 of this)

File: src/design-system/InputLabelAndHelp.tsx (hunk 3 above)
Comment (szhu) [reply src/design-system/InputLabelAndHelp.tsx:26]: Also can you call this `labelHelpPlacement`?

File: src/page-response-impact/RtSummaryStats.tsx [hunk 5]
```diff
@@ -139,6 +141,10 @@ const RtSummaryStats: React.FC<Props> = ({ rtData }) => {
         <LegendContainer>
           <LegendSpan />
           <div>{`Facilities with R(t) < 1.0`}</div>
+          <Spacer x={5} />
+          <HelpButtonWithTooltip placement="top">
+            {`An R(t) < 1.0 means the virus will stop spreading. An R(t) > 1.0 means the virus will spread quickly.`}
```
Comment (szhu) [inline src/page-response-impact/RtSummaryStats.tsx:146]: Note: Another acceptable solution:

```suggestion
            An R(t) &lt; 1.0 means the virus will stop spreading. An R(t) &gt; 1.0 means the virus will spread quickly.
```

I think this is better than the one you chose in some aspects and worse in some other aspects. Up to you which to choose!

**Comment by amyrlam** [comment]
this was added on the side with https://github.com/Recidiviz/covid19-dashboard/pull/366

i still need to figure this out, and will open a new branch and tag you @szhu when get to that

**Pull request closed without merging**